.github/workflows/e2e-tests-workflow.yml

```diff
@@ -0,0 +1,74 @@
+name: E2E tests workflow
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  tests:
+    name: Run e2e tests
+    strategy:
+      matrix:
+        os: [ubuntu-16.04] # use ubuntu-16.04 as required by cypress: https://github.com/marketplace/actions/cypress-io#important
+        java: [14]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Pull and Run Docker
+        run: |
+          version=1.8.0
+          echo $version
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            ## The ESRestTest Client uses http by default.
+            ## Need to disable the security plugin to call the rest api over http.
+            echo "RUN if [ -d /usr/share/elasticsearch/plugins/opendistro_security ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security; fi" >> Dockerfile
+            docker build -t odfe-ad:test .
+          fi
+      - name: Run Docker Image
+        run: |
+          docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" odfe-ad:test
+          sleep 90
+          curl -XGET http://localhost:9200/_cat/plugins
+
+      - name: Checkout Kibana
+        uses: actions/checkout@v2
+        with:
+          repository: elastic/kibana
```

Comment:
Let's hold for some time for response of using `elastic/kibana` or kibana oss. Similar to this one https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/207